core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/user/AuthenticateUser.kt

```diff
@@ -0,0 +1,54 @@
+package org.mifos.mobilewallet.core.domain.usecase.user
+
+import org.mifos.mobilewallet.core.base.UseCase
+import org.mifos.mobilewallet.core.data.fineract.entity.UserEntity
+import org.mifos.mobilewallet.core.data.fineract.entity.authentication.AuthenticationPayload
+import org.mifos.mobilewallet.core.data.fineract.entity.mapper.UserEntityMapper
+import org.mifos.mobilewallet.core.data.fineract.repository.FineractRepository
+import org.mifos.mobilewallet.core.domain.model.user.User
+import org.mifos.mobilewallet.core.utils.Constants
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by naman on 16/6/17.
+ */
+class AuthenticateUser @Inject constructor(private val apiRepository: FineractRepository) :
+    UseCase<AuthenticateUser.RequestValues, AuthenticateUser.ResponseValue>() {
+
+    @JvmField
+    @Inject
+    var userEntityMapper: UserEntityMapper? = null
```

Comment:
You are injecting, it means it is already initialized in hilt graph then It should not be nullable remove `? = null`